src/services/AbstractDFG.jl

```diff
@@ -472,14 +516,16 @@ Notes:
 - used by both factor graph variable and Bayes tree clique logic.
 """
 function isInitialized(var::DFGVariable; key::Symbol=:default)::Bool
-  solverData(var, key) != nothing && return solverData(var, key).initialized
-  return false
-end
-function isInitialized(fct::DFGFactor; key::Symbol=:default)::Bool
-  solverData(var, key) != nothing && return solverData(fct, key).initialized
-  return false
+  	data = solverData(var, key)
+  	if data == nothing
+		@error "Variable does not have solver data $(key)"
```

Comment:
I thought an @error should be raised, because something is seriously wrong and a false return just lets me think its not initialized.

Reply:
just making sure you are confortable with difference between `error(..)` and `@error`?

Reply:
haha, I do. We should not be using error() in any case, it throws an generic error.

Reply:
Didn't want to mess with the existing behavior as it is a breaking change.

Reply:
cool, so to be sure -- how do you capture `@error` calls?

Reply:
You don’t, user reads in REPL or log.